src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -750,5 +733,311 @@ private static string ToHtmlCase(string name)
+                    _index = nameEndIndex;
+
+                    hasWildcard = _requiredAttributes[nameEndIndex] == RequiredAttributeWildcardSuffix;
+                    if (hasWildcard)
+                    {
+                        // Move past wild card
+                        Next();
+                    }
+                }
+
+                TagHelperRequiredAttributeDescriptor descriptor = null;
+                if (ValidateName(attributeName, targetingAttributes: true, errorSink: errorSink))
+                {
+                    descriptor = new TagHelperRequiredAttributeDescriptor
+                    {
+                        Name = attributeName
+                    };
+
+                    if (hasWildcard)
+                    {
+                        descriptor.Operator = RequiredAttributeWildcardSuffix;
+                    }
+                }
+
+                return descriptor;
+            }
+
+            private string ParseCSSAttributeName(ErrorSink errorSink)
+            {
+                var nameStartIndex = _index;
+                var nameEndIndex = _requiredAttributes.IndexOfAny(new char[] { ' ', '\t', ']', '=', '^', '$' }, _index);
+                nameEndIndex = nameEndIndex == -1 ? _requiredAttributes.Length : nameEndIndex;
+                _index = nameEndIndex;
+
+                var attributeName = _requiredAttributes.Substring(nameStartIndex, nameEndIndex - nameStartIndex);
+
+                return attributeName;
+            }
+
+            private char ParseCSSValueOperator(ErrorSink errorSink)
+            {
+                Debug.Assert(!AtEnd);
+
+                if (TagHelperRequiredAttributeDescriptor.SupportedCSSValueOperators.Contains(Current))
+                {
+                    var op = Current;
+                    Next();
+
+                    if (op != '=' && At('='))
+                    {
+                        // Two length operator (ex: ^=). Move past the second piece
+                        Next();
+                    }
+
+                    return op;
+                }
+
+                return '\0';
+            }
+
+            private string ParseCSSValue(ErrorSink errorSink)
+            {
+                int valueStart, valueEnd;
+                if (At('\'') || At('"'))
+                {
+                    var quote = Current;
+
+                    // Move past the quote
+                    Next();
+
+                    valueStart = _index;
+                    var quoteEndIndex = _requiredAttributes.IndexOf(quote, _index);
+                    if (quoteEndIndex == -1)
+                    {
+                        errorSink.OnError(
+                            SourceLocation.Zero,
+                            Resources.FormatTagHelperDescriptorFactory_InvalidRequiredAttributeCharacterExpectedOther(
+                                _requiredAttributes[_requiredAttributes.Length - 1],
+                                quote),
```

Comment:
Lousy message for this case. Say something about an unclosed attribute value or string literal or something.